tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,259 @@
+import { FilterSearch } from '../../src/components/FilterSearch';
+import { render, screen } from '@testing-library/react';
+import * as useSynchronizedRequestFunctions from '../../src/hooks/useSynchronizedRequest';
+import userEvent from '@testing-library/user-event';
+import * as searchOperations from '../../src/utils/search-operations';
+import { mockAnswersActions } from '../__utils__/mocks';
+import { sectionedFilterSearchResponse, unsectionedFilterSearchResponse, noResultsFilterSearchResponse} from '../../tests/__fixtures__/data/filtersearch';
+
+jest.mock('@yext/answers-headless-react');
+
+const mockedExecuteFilterSearch = jest.fn();
+const setFilterOption = jest.fn();
+const setOffset = jest.fn();
+const searchFieldsProp = [{
+  fieldApiName: 'name',
+  entityType: 'ce-person'
+}];
+
+describe('search with section labels', () => {
+  beforeEach(() => {
+    mockAnswersActions({
+      setFilterOption,
+      setOffset
+    });
+
+    jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+    jest.spyOn(useSynchronizedRequestFunctions, 'useSynchronizedRequest').mockImplementation(() => {
+      return [sectionedFilterSearchResponse, mockedExecuteFilterSearch, null];
+    });
+  });
+
+  it('renders the filter search bar and the "Filter" label', () => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const label = 'Filter';
+    const labelElement = screen.getByText(label);
+    const searchBarElement = screen.getAllByRole('textbox');
+
+    expect(labelElement).toBeDefined();
+    expect(searchBarElement.length).toBe(1);
+  });
+
+  it('calls executeFilterSearch every time a character is typed or backspaced', () => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, 'n');
+    expect(searchBarElement).toHaveValue('n');
+    expect(mockedExecuteFilterSearch).toHaveBeenLastCalledWith('n');
+
+    userEvent.type(searchBarElement, 'a');
+    expect(searchBarElement).toHaveValue('na');
+    expect(mockedExecuteFilterSearch).toHaveBeenLastCalledWith('na');
+
+    userEvent.type(searchBarElement, '{backspace}');
+    expect(searchBarElement).toHaveValue('n');
+    expect(mockedExecuteFilterSearch).toHaveBeenLastCalledWith('n');
+
+    expect(mockedExecuteFilterSearch).toHaveBeenCalledTimes(3);
+  });
+
+  it('does not trigger executeFilterSearch when backspacing on an empty text box', () => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, '{backspace}');
+    expect(searchBarElement).toHaveValue('');
+
+    expect(mockedExecuteFilterSearch).toHaveBeenCalledTimes(0);
+  });
+
+  it('shows autocomplete suggestions when a character is typed and suggestions are returned by useSynchronizedSearch', () => {
```

Comment:
I'd probably remove the `useSynchronizedSearch` since that's a bit on the implementation detail side of things.

Reply:
just did, one step ahead of ya! haha